ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java

```diff
@@ -263,7 +261,7 @@ private MemoizedSupplier<List<CompletableFuture<Message>>> applyLog() throws Raf
         final long incremented = appliedIndex.incrementAndGet(debugIndexChange);
         Preconditions.assertTrue(incremented == nextIndex);
         if (f != null) {
-          futures.get().add(f);
+          applyLogFutures = applyLogFutures.thenCombine(f, (previous, current) -> previous);
```

Comment:
It should return `current`, i.e. `(previous, current) -> current`.

Reply:
current is a Message class returned and previous is null value Void

Reply:
We don't need the Message returned after apply transactionLog

Reply:
You are right.  Then, let's return null, i.e. `(v, message) -> null`.

Reply:
@swamirishi , any update for this?